### DISCUSSION TITLE
How to compress a file and copy to a couple of folders ?

### DISCUSSION THREAD

**Discussion by Izanatos** [drakkan/sftpgo#1252] [Q&A]
Hi guys,

I've been fiddling for hours with the actions but I am totally out of my depth.

I am trying to compress and copy the file to a couple of locations after a upload is completed.

I am running the server on windows.

After trying all sorts of combinations I've only achieved to compress a folder.

![image](https://user-images.githubusercontent.com/46830654/228658329-4d9278dd-ca8f-436a-8728-66d87069a39c.png)

the root for user1 is C;/sftp/user1/

a file is being created when I upload a file but I'am only getting a corrupted filepart.
![image](https://user-images.githubusercontent.com/46830654/228658671-6274a8c9-4838-465d-a1ae-f6c2b6025355.png)

any help would be much appreciated,

Thanks,
A

**Comment by drakkan** [comment]
Hello,

I guess you are using WinSCP as client. It uploads files by appending the `.filepart` suffix to the filename and then renames the file once the upload is finished. You have to write the rules/actions considering how your client works. 

I'm sorry but I no longer provide free step-by-step support, please check the [support policy](https://github.com/drakkan/sftpgo#support-policy). Thank you

**Comment by JonBendtsen** [comment]
You are correct to use an Event action.

I would suggest that you create a script on your SFTPGo server that does what you want, and then in uses the event manager to call this script. I've done something similar where my script scan a file using clamav and then I copy the clean files to one directory and the dirty files to a different directory. The script is started by SFTPGo on all uploaded files and parameters like the uploaded filename is read by the script from environment variables.

All mine is done inside containers running on Linux, but the principals should be the same on Windows.

I would still strongly suggest that you support Drakkans work as pr. the [support-policy](https://github.com/drakkan/sftpgo#support-policy)
